apps/web/src/components/BridgeModal/BridgeForm.tsx

```diff
@@ -0,0 +1,222 @@
+import { useConnectModal } from '@rainbow-me/rainbowkit'
+import { sendTransaction } from '@wagmi/core'
+import { Box, Button, Flex, Heading, Text } from '@zoralabs/zord'
+import { parseEther } from 'ethers/lib/utils.js'
+import { Formik } from 'formik'
+import Image from 'next/image'
+import { useState } from 'react'
+import { useAccount, useBalance, useNetwork, useSwitchNetwork } from 'wagmi'
+
+import Input from 'src/components/Input/Input'
+import { L2ChainType, PUBLIC_L1_BRIDGE_ADDRESS } from 'src/constants/addresses'
+import { PUBLIC_DEFAULT_CHAINS } from 'src/constants/defaultChains'
+import { useBridgeModal } from 'src/hooks/useBridgeModal'
+import { useIsContract } from 'src/hooks/useIsContract'
+import { useChainStore } from 'src/stores/useChainStore'
+import { formatCryptoVal } from 'src/utils/numbers'
+
+import { Icon } from '../Icon'
+import bridgeFormSchema, { BridgeFormValues } from './BridgeForm.schema'
+import { NetworkSelector } from './NetworkSelector'
+
+export const BridgeForm = () => {
+  const { address } = useAccount()
+  const { chain: userChain } = useNetwork()
+  const { switchNetwork } = useSwitchNetwork()
+  const { closeBridgeModal } = useBridgeModal()
+  const { openConnectModal } = useConnectModal()
+  const { data: isContractWallet } = useIsContract({ address })
+  const [loading, setLoading] = useState(false)
+
+  const { chain: appChain } = useChainStore()
+
+  const l1Chain = PUBLIC_DEFAULT_CHAINS[0]
+  const [l2Chain, setL2Chain] = useState(
+    appChain.id !== l1Chain.id ? appChain : PUBLIC_DEFAULT_CHAINS[1]
+  )
+
+  const isWalletOnL1 = userChain?.id === l1Chain.id
+
+  const { data: userL1Balance } = useBalance({
+    address,
+    chainId: l1Chain.id,
+  })
+
+  const { data: userL2Balance } = useBalance({
+    address,
+    chainId: l2Chain.id,
+  })
+
+  const initialValues: BridgeFormValues = {
+    amount: 0,
+  }
+
+  const handleSubmit = async (values: BridgeFormValues) => {
+    const bridge = PUBLIC_L1_BRIDGE_ADDRESS[l2Chain.id as L2ChainType]
+
+    if (!values.amount || !bridge) return
+
+    setLoading(true)
+    try {
+      const { wait } = await sendTransaction({
+        request: {
+          to: PUBLIC_L1_BRIDGE_ADDRESS[l2Chain.id as L2ChainType],
+          value: parseEther(values.amount.toString()),
+        },
+        mode: 'recklesslyUnprepared',
+      })
+      await wait()
+    } catch (err) {
+      console.log('err', err)
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const formattedL1Balance = userL1Balance ? parseFloat(userL1Balance.formatted) : 0
+  const formattedL2Balance = userL2Balance ? parseFloat(userL2Balance.formatted) : 0
+
+  const getButtonText = (isAmountInvalid: boolean) => {
+    if (isContractWallet) return 'Contract wallets are not supported'
```

Comment:
oh interesting - what is a contract wallet?

Reply:
oooh i can take this one! 😀😀

examples: multisig wallet or treasury addresses that aren't EOAs

Reply:
to add to @isaaczaak response EOAs are guaranteed to have the same address across any evm chain where as a smart contract wallet is not. for this bridge interface since we are using the optimism portal to send eth to the same account on L2 we are restricting to only EOAs for now.